backend/hct_mis_api/apps/account/models.py

```diff
@@ -379,30 +382,25 @@ def cached_has_user_roles_for_business_area_and_permission(
     def has_permission(
         self, permission: str, business_area: BusinessArea, program_id: Optional[UUID] = None, write: bool = False
     ) -> bool:
-        has_program_access = True
-        has_partner_roles = False
-
-        if not self.partner.is_unicef:
-            if program_id:
-                has_program_access = str(program_id) in self.get_partner_programs_areas_dict(
-                    business_area_id=business_area.pk
-                )
-            has_partner_roles = self.cached_has_partner_roles_for_business_area_and_permission(
-                business_area=business_area,
-                permission=permission,
-            )
-
         has_user_roles = self.cached_has_user_roles_for_business_area_and_permission(
             business_area=business_area,
             permission=permission,
         )
 
         if self.partner.is_unicef:
-            return has_program_access and (
-                has_user_roles or permission in DEFAULT_PERMISSIONS_LIST_FOR_IS_UNICEF_PARTNER
-            )
-        else:
-            return has_program_access and (has_user_roles or has_partner_roles)
+            return has_user_roles
```

Comment:
```
if self.partner.is_unicef:
    has_program_access = True
    has_partner_roles = permission in DEFAULT_PERMISSIONS_LIST_FOR_IS_UNICEF_PARTNER
else:
    has_program_access = str(program_id) in self.get_partner_programs_areas_dict(business_area_id=business_area.pk)
    has_partner_roles = self.cached_has_partner_roles_for_business_area_and_permission(
            business_area=business_area,
            permission=permission,
        )
```
?
Permission can still originate from partner UNICEF

Reply:
UNICEF partner still has access to all programs, but user permissions determine to which business area and which features, in previous implementations in some cases, UNICEF partner had access to all business areas

Reply:
But is it supposed to no longer grant any permissions now? Because right now users under UNICEF partner will not inherit any permissions.
Shouldn't UNICEF partner grant the permissions from the default list for the BAs that user has access to? So if user has access to Afg, but with role that grants only some 1 basic permission, he should have all permissions from UNICEF partner inside this BA, I think.